web_api/national_park/api.py

```diff
@@ -0,0 +1,55 @@
+# This section imports necessary libraries requests , os , logging and pprint.
+
+import os
+import requests
+from typing import List
+from park import Park
+
+base_url = 'https://developer.nps.gov/api/v1' # Sets a variable base_url to the base URL for the National Park Service API.
+
+# The get_parks_data function is called with the specified search criteria. 
+# This function constructs a URL with the API endpoint and parameters, 
+# and sends a GET request to retrieve data from the API. The response data is converted to JSON format 
+# and the "data" field is extracted as a list of dictionaries.
+
+def get_parks_data(state: str, search_query: str = "", park_code: str = None, limit: int = 5) -> List[dict]:
+    url = f"{base_url}/parks"
+    api_key = os.getenv('NPS_API_KEY') # a GET request to the National Park Service API.
```

Comment:
we should move the API key to be a global variable. so when we start adding new functions that use different API end points we don't need to repeat this line